## Mean median mode activity

In [9]:
import statistics

def get_mean_med_mode(data):    
    print('Mean: ', statistics.mean(data))
    print('Mode: ', statistics.mode(data))
    print('Median: ', statistics.median(data))


annual_incomes = [45, 52, 38, 90, 45, 36, 70, 100, 85, 6]
print('\nIncomes: ')
get_mean_med_mode(annual_incomes)

light_hours = [2090, 1900, 2010, 1890, 2010, 1968, 70, 2200, 2012, 1960]
print('\nLight Hours:')
get_mean_med_mode(light_hours)






Incomes: 
Mean:  56.7
Mode:  45
Median:  48.5

Light Hours:
Mean:  1811
Mode:  2010
Median:  1989.0


In [13]:
height_counts = [13 ,215,1459,3413,3243,1512,135,10]
half_way = sum(height_counts[:4])
print(half_way, height_counts[4])
sum(height_counts)

# mode = guess that likely in the 64 - 82 range but can't know for sure
# median = somewhere in the range of 64-70 as this is where the count passes 5000
# mean = calc weighted means for each height range = middle of range times whatever get something



5100 3243


10000

## 1.14 Percentiles and quartiles

In [18]:
data = [12, 16, 20, 25, 26, 28, 33, 35, 36, 41, 44, 49, 53, 56, 61, 63, 67, 70, 75, 77]

percentiles = statistics.quantiles(data, n=100)
search_percentiles = [10, 25, 33, 50, 75, 90]

#use p-1 because 0 indexed

for p in search_percentiles:
    print(p, percentiles[p-1])


10 16.4
25 26.5
33 32.65
50 42.5
75 62.5
90 74.5


## 1.17 Variance



In [34]:
import numpy as np

data = [45, 52, 38, 90, 45, 36, 70, 100, 85, 60]

statistics.stdev(data)
#22.98526097790108

statistics.variance(data)
#528.3222222222222


elon_data = data + [1000000000]
print(elon_data)

print('elon standard deviation:', statistics.stdev(elon_data))
# 301511325

print('elon variance: ', statistics.variance(elon_data))
# 9.090907961818264e+16

cv = cv = lambda x: np.std(x, ddof=1) / np.mean(x) * 100 

print('coeffient of variation without elon', cv(data))
print('coeffient of variation with elon', cv(elon_data))





[45, 52, 38, 90, 45, 36, 70, 100, 85, 60, 1000000000]
elon standard deviation: 301511325.8539099
elon variance:  9.090907961818264e+16
coeffient of variation without elon 37.01330270193411
coeffient of variation with elon 331.6622524770421


Compute the variance and standard deviation of the data.

- Standard deviation: 22.985
- Variance = 528.32


Now, assume that this data is a sample chosen scientifically to help us understand a larger group of pub goers. How do your answers change?

- No this does not change anytihng.


Along comes Elon Musk with his 14 billion pound annual income. Recompute the values from question 1 to include Elon's income. 

elon standard deviation: 301511325.8539099
elon variance:  9.090907961818264e+16

Question 4

Compute the coefficient of variation for the data with and without Elon's income.

coeffient of variation without elon 37.01330270193411
coeffient of variation with elon 331.6622524770421